src/Behat/Config/Profile.php

```diff
@@ -30,6 +30,13 @@ public function name(): string
         return $this->name;
     }
 
+    public function withExtension(ExtensionConfigInterface $extension): self
+    {
+        $this->settings['extensions'][$extension->name()] = $extension->toArray();
```

Comment:
Shouldn't we check here that there is not already a config for this extension in the same way that we checked that there was not a duplicate config for a suite?

Reply:
Good catch!

Reply:
done